prime-dotnet-webapi/Services/AuthorizedUserService.cs

```diff
@@ -0,0 +1,137 @@
+                .SingleOrDefaultAsync();
+        }
+
+        public async Task<AuthorizedUserViewModel> GetAuthorizedUserForUserIdAsync(Guid userId)
+        {
+            return await GetBaseAuthorizedUserQuery()
+                .Where(au => au.Party.UserId == userId)
+                .ProjectTo<AuthorizedUserViewModel>(_mapper.ConfigurationProvider)
+                .SingleOrDefaultAsync();
+        }
+
+        public async Task<int> CreateOrUpdateAuthorizedUserAsync(AuthorizedUserChangeModel changeModel, ClaimsPrincipal user)
+        {
+            var authorizedUser = await GetBaseAuthorizedUserQuery()
+                .SingleOrDefaultAsync(au => au.Party.UserId == user.GetPrimeUserId());
+
+            if (authorizedUser == null)
+            {
+                var party = new Party
+                {
+                    Addresses = new List<PartyAddress>()
+                };
+                _context.Parties.Add(party);
+
+                authorizedUser = new AuthorizedUser
+                {
+                    Party = party,
+                    Status = AccessStatusType.UnderReview
+                };
+                _context.AuthorizedUsers.Add(authorizedUser);
+            }
+
+            changeModel.UpdateAuthorizedUser(authorizedUser, user);
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                return InvalidId;
+            }
+
+            return authorizedUser.Id;
+        }
+
+        public async Task ActivateAuthorizedUser(int authorizedUserId)
+        {
+            var authorizedUser = await _context.AuthorizedUsers
+                .SingleOrDefaultAsync(au => au.Id == authorizedUserId);
+
+            authorizedUser.Status = AccessStatusType.Active;
+
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task ApproveAuthorizedUser(int authorizedUserId)
+        {
+            var authorizedUser = await _context.AuthorizedUsers
+                .SingleOrDefaultAsync(au => au.Id == authorizedUserId);
+
+            authorizedUser.Status = AccessStatusType.Approved;
+
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task DeleteAuthorizedUserAsync(int authorizedUserId)
+        {
+            var authorizedUser = await _context.AuthorizedUsers
+                .SingleOrDefaultAsync(au => au.Id == authorizedUserId);
+
+            if (authorizedUser == null)
+            {
+                return;
+            }
+
+            await _partyService.DeletePartyAsync(authorizedUser.Party.Id);
+        }
+
+        private IQueryable<AuthorizedUser> GetBaseAuthorizedUserQuery()
+        {
+            return _context.AuthorizedUsers
+                .Include(au => au.Party)
+                    .ThenInclude(p => p.Addresses)
+                        .ThenInclude(pa => pa.Address)
+                .Include(au => au.HealthAuthority);
```

Comment:
I don't think `HealthAuthority` is ever used (as opposed to `HealthAuthorityCode`)? I don't think you need this `Include()`.

Reply:
Yah definitely don't.  It would be used if I hadn't stashed that endpoint for the next iteration of health authorities.  I'll add this to my stash.